straxen/misc.py

```diff
@@ -276,3 +283,73 @@ def convert_array_to_df(array: np.ndarray) -> pd.DataFrame:
     """
     keys = [key for key in array.dtype.names if array[key].ndim == 1]
     return pd.DataFrame(array[keys])
+
+
+@export
+class CacheDict(OrderedDict):
+    """Dict with a limited length, ejecting LRUs as needed.
+        copied from
+        https://gist.github.com/davesteele/44793cd0348f59f8fadd49d7799bd306
+    """
+
+    def __init__(self, *args, cache_len: int = 10, **kwargs):
+        assert cache_len > 0
+        self.cache_len = cache_len
+
+        super().__init__(*args, **kwargs)
+
+    def __setitem__(self, key, value):
+        super().__setitem__(key, value)
+        super().move_to_end(key)
+
+        while len(self) > self.cache_len:
+            oldkey = next(iter(self))
+            super().__delitem__(oldkey)
+
+    def __getitem__(self, key):
+        val = super().__getitem__(key)
+        super().move_to_end(key)
+
+        return val
+
+@export
+def total_size(o, handlers={}, verbose=False):
```

Comment:
for good practice, can you not pass mutables as defaults?

https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/

Reply:
done.